zson/formatter.go

```diff
@@ -422,3 +422,74 @@ func (t typemap) known(typ zng.Type) bool {
 	}
 	return typ.ID() < zng.IdTypeDef
 }
+
+// FormatType formats a type in canonical form to represent type values
+// as standalone entities.
+func FormatType(typ zng.Type) string {
+	var b strings.Builder
+	formatType(&b, make(typemap), typ)
+	return b.String()
+}
+
+func formatType(b *strings.Builder, typedefs typemap, typ zng.Type) {
+	if name, ok := typedefs[typ]; ok {
+		b.WriteString(name)
+		return
+	}
+	switch t := typ.(type) {
+	case *zng.TypeAlias:
+		name := t.Name
+		b.WriteString(name)
+		if _, ok := typedefs[typ]; !ok {
+			typedefs[typ] = name
+			b.WriteString("=(")
+			formatType(b, typedefs, t.Type)
+			b.WriteByte(')')
+		}
+	case *zng.TypeRecord:
+		b.WriteByte('{')
+		var sep string
+		for _, col := range t.Columns {
+			b.WriteString(sep)
```

Comment:
Nit: I like this pattern for the separator here and for enums and unions, but what's here is just fine!
```go
		for k, col := range t.Columns {
			if k > 0 {
				b.WriteByte(',')
			}
```